includes/html/modal/new_bill.inc.php

```diff
@@ -44,12 +44,18 @@
                         <select class="form-control input-sm" id="device" name="device" onchange="getInterfaceList(this)">
                             <option value=''>Select a device</option>
                             <?php
-                              $devices = dbFetchRows('SELECT * FROM `devices` ORDER BY hostname');
-                            foreach ($devices as $device) {
-                                $selected = $device['device_id'] == $port_device_id ? " selected" : "";
-                                echo "<option value='${device['device_id']}' $selected>${device['hostname']}</option>\n";
-                            }
-                                ?>
+                              $sort_field = 'hostname';
+
+                              if (\LibreNMS\Config::get('force_ip_to_sysname') === true) {
+                                  $sort_field = 'sysName';
+                              }
+
+                              $devices = dbFetchRows('SELECT * FROM `devices` ORDER BY ?', array($sort_field));
```

Comment:
This actually won't sort correctly if the user mixes dns named and ip hosts.

You need to fetch the devices then sort them in PHP.

Honestly, you should use select2 for these.  Search for init_select2 in the code to see how.

Reply:
> This actually won't sort correctly if the user mixes dns named and ip hosts.

That's how it is now though so this doesn't regress anything just honours an existing config option.

Reply:
I thought so too laf.

Reply:
Yeah, but if you use the select2 code, the device names will respect the setting, be sorted properly and not blow up if there are 100,000 devices or something.

Yes, code is fine as is.

Really all you need is this: 
```php
<select class="form-control input-sm" id="device" name="device" onchange="getInterfaceList(this)"></select>
<script type="text/javascript">
    init_select2('#device', 'device', {}, <?php echo "{id: $port_device_id, text: '" . format_hostname($device) . "'}");
</script>
```

(You'll have to get $device from somewhere to find the hostname)